zerver/views/message_summary.py

```diff
@@ -0,0 +1,131 @@
+import json
+from typing import Any
+
+import litellm
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from django.utils.timezone import now as timezone_now
+from pydantic import Json
+
+from analytics.lib.counts import COUNT_STATS, do_increment_logging_stat
+from zerver.lib.narrow import NarrowParameter
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import UserProfile
+from zerver.views.message_fetch import get_messages_backend
+
+# Maximum number of messages that can be summarized in a single request.
+MAX_MESSAGES_SUMMARIZED = 100
+OUTPUT_COST_PER_TOKEN = 5
+INPUT_COST_PER_TOKEN = 1
```

Comment:
Where do these last two numbers come from? I think it's worth having comments explaining how they were computed here. (Maybe we'll find it best to have them live in configuration, alongside the choice of model, so that may move, but it seems important)

Reply:
(I think for most applications, the input and output token costs seem to be the same).

Reply:
I guess maybe they'd just sample values? Anyway I think what we want to be using is something like the multiplier of $1/1B times the per-token price, if that's a reasonable unit to use. (What does that become for Bedrock?)

Reply:
Yeah, these are just sample values. 

As per bedrock [pricing page](https://aws.amazon.com/bedrock/pricing/?refid=33b5edcf-0d26-4e1d-b868-603c42c06eaf):

![image](https://github.com/user-attachments/assets/8eadb89a-8fe3-4fb3-bd69-a77076cbab65)

Should we allow credits_used to be decimal here? Right now it can only have integer values:
```
class BaseCount(models.Model):
    # Note: When inheriting from BaseCount, you may want to rearrange
    # the order of the columns in the migration to make sure they
    # match how you'd like the table to be arranged.
    property = models.CharField(max_length=32)
    subgroup = models.CharField(max_length=16, null=True)
    end_time = models.DateTimeField()
    value = models.BigIntegerField()
```

Reply:
No, we need to work with integrates. Which is why I was doing looking at a translation of the form:
- `$0.00072/1K tokens = 720 in units of ($1/1B tokens), so we'd put 720 for both input and output here, and a comment explaining how we came to that.

Reply:
Makes sense. Done.